### ISSUE TITLE
Dispose() not closing bound ports?

### ISSUE THREAD

**Issue by JSONVoorhees** [statianzo/Fleck#243]
Hi, I'm over on Linux (Ubuntu 18.04 LTS) using .NET Core 2.1.301. 

I'm hitting an issue where when I use Ctrl-C to shut down my server during routine testing, it can take upwards of a minute for the port Fleck has bound to release, which results in an exception (`System.Net.Sockets.SocketException (98): Address already in use`) when the server is started up again within that window.

I've attempted to use IApplicationLifeTime to register a call to Dispose() when the application has reached a stopped state, which seems to work within the first few seconds of the server being live if I Ctrl-C and then restart it. However, once the server has been up and running for a bit, it will not release the port cleanly. 

As documentation is sparse, I'm wondering if there's something I'm missing here. I've attempted to look through the issues here but did not see something pertinent to this problem. Thanks for your help!

```
    public class Startup
    {
        public void Configure(IApplicationBuilder app, IApplicationLifetime appLifetime)
        {
            var connected = new List<IWebSocketConnection>();
            var socketsrv = new WebSocketServer("ws://127.0.0.1:8181");
            appLifetime.ApplicationStopped.Register(socketsrv.Dispose);
            socketsrv.Start(socket =>
            {
                socket.OnOpen = () => WebSockets.OnOpen(socket, connected);
                socket.OnClose = () => WebSockets.OnClose(socket, connected);
                socket.OnMessage = (message) => WebSockets.OnMessage(socket, connected, message);
            });

            app.UseOwin(x => x.UseNancy());
        }
    }
```

**Comment by JSONVoorhees** [comment]
As a follow-up, I've also attempted: 

```
            appLifetime.ApplicationStopping.Register(() =>
            {
                foreach (var socket in connected) socket.Close();
                socketsrv.Dispose();
            });
```

Unfortunately, this did not do the trick either.

**Comment by AdrianBathurst** [comment]
In my app, I'm doing just like you're doing...close all existing connections then call `socketsrv.Dispose();` and it works fine....though this is on Windows Server. Debug and make sure this portion of the code is being triggered.

**Comment by JSONVoorhees** [comment]
Yes, I tested this initially but did not mention it. The function called by `ApplicationStopped.Register` always fires.

```
            appLifetime.ApplicationStopped.Register(TestFunc);
[...]
        private void TestFunc()
        {
            Console.WriteLine("Working!");
        }
```

Result:
```
Application started. Press Ctrl+C to shut down.
^CApplication is shutting down...
Working!
```

**Comment by JSONVoorhees** [comment]
I've also tested hooking both `ApplicationStopping` and `ApplicationStopped`, hence the discrepancy above.

**Comment by JSONVoorhees** [comment]
Had some time to further investigate. It appears that when a Fleck server is spun up and a websocket connection is established, if you do not close the browser tab prior to restarting the server, it will place the established connection into a TIME_WAIT state when the server is shut down:

```Shell
# Server running, connection established.
$ netstat -ap | grep 8181
tcp        0      0 localhost:8181          0.0.0.0:*               LISTEN      20160/./server      
tcp        0      0 localhost:40562         localhost:8181          ESTABLISHED 19388/chromium-brow 
tcp        0      0 localhost:8181          localhost:40562         ESTABLISHED 20160/./server      

# After shutting the server down.
$ netstat -ap | grep 8181
tcp        0      0 localhost:8181          localhost:40562         TIME_WAIT   -     
```
Until the TIME_WAIT timeout has elapsed which defaults to two minutes, the Fleck server will throw an exception when restarting.

**Comment by JSONVoorhees** [comment]
Looks like the issue is related to https://github.com/dotnet/corefx/issues/24562

**Comment by JSONVoorhees** [comment]
This issue just got re-broken by the recent set of commits. 

`socket.SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.ReuseAddress, 1);` is now gated behind `FleckRuntime.IsRunningOnWindows()` rather than also running on Linux platforms as needed.
